webauthn/webauthn.py

```diff
@@ -175,8 +177,26 @@ def json(self):
 
 
 class WebAuthnUser(object):
-    def __init__(self, user_id, username, display_name, icon_url,
-                 credential_id, public_key, sign_count, rp_id):
+    def __init__(self, *args, **kwargs):
```

Comment:
Can we just leave the method definition as-is, then raise if a necessary value is missing?

Reply:
The problem would be if not all arguments are set, like in :

```
WebAuthnUser(credential_id = "value")
TypeError: __init__() takes exactly 9 arguments (2 given)
```

Would be possible to set all parameters to None, and check them after :

```
def __init__(self, user_id=None, username=None, display_name=None, icon_url=None,
                 credential_id=None, public_key=None, sign_count=None, rp_id=None)
```
But I dont know if it's more readable.

Reply:
I think it's more idiomatic to leave the expected parameters as positional arguments. The `TypeError` signals to the caller that 9 arguments are expected, which they are. It makes sense to make the optional arguments keyword arguments, and set their default values to `None`, and leave the required arguments as positional. The problem with doing this, however, is that we risk making a non-backwards compatible API change. I think at this point it'd be better to leave the method signature as-is, (except for adding new keyword arguments). If we need to add a new positional argument in the future I think at that point it would make sense to rework the positional arguments and make non-essential ones `kwargs`, (maybe `icon_url`?). For now we can just raise if a required positional argument is empty, (`None` or `''`).

Reply:
With the current signature :
```
def __init__(self, user_id, username, display_name, icon_url,
                 credential_id, public_key, sign_count, rp_id):
```
and not changing it to a more flexible one, for example :

```
def __init__(self, user_id=None, username=None, display_name=None, icon_url=None,
                 credential_id=None, public_key=None, sign_count=None, rp_id=None):
```
we will have to provide anyway the non-mendatory data like this :

```
webauthn_user = WebAuthnUser( 
    None, None, None, None,
    "credential_id", None, None, "rp_id"
)
```
which was the main argument of the PR.